src/chexus/validators.py

```diff
@@ -257,6 +257,31 @@ def validate(self, node: Dataset | Group) -> Violation | None:
         return Violation(node.name)
 
 
+class event_id_subset_of_detector_number(Validator):
+    def __init__(self) -> None:
+        super().__init__(
+            "event_id is not subset of associated detector_numbers",
+            "The values in the event_id field in NXevent_data should be "
+            "a subset of the values in the detector_number dataset on the "
+            "associated NXdetector.",
+        )
+
+    def applies_to(self, node: Dataset | Group) -> bool:
+        return (
+            isinstance(node, Group)
+            and node.attrs.get('NX_class') == 'NXevent_data'
+            and 'event_id' in node.children
+            and 'detector_number' in node.parent.children
```

Comment:
Are we separately checking that `event_id` and `detector_number` exist? Or is it ok when they don't and we can load the file anyway?

Reply:
We don't have validators that makes sure `event_id` and `detector_number` exists at the moment. I don't know if that's a requirement. Maybe we should create an issue for that.